src/js/actions/documents.js

```diff
@@ -669,6 +669,8 @@ define(function (require, exports) {
                 interactionMode: descriptor.interactionMode.DISPLAY
             };
 
+        this.transfer(toolActions.changeVectorMaskMode, false);
```

Comment:
We still need to block the close action from completing until this transfer finishes. My previous nit was that we were executing this sequentially and only after everything else had finished. Instead we should execute transfers as early as possible and with as much concurrency as possible, but still block the action until all have completed.